src/Microsoft.AspNetCore.Mvc.Core/Builder/MvcApplicationBuilderExtensions.cs

```diff
@@ -85,6 +87,51 @@ public static IApplicationBuilder UseMvcWithDefaultRoute(this IApplicationBuilde
                     "ConfigureServices(...)"));
             }
 
+            var middlewarePipelineBuilder = app.ApplicationServices.GetRequiredService<MiddlewarePipelineBuilderService>();
+            middlewarePipelineBuilder.ApplicationBuilder = app;
```

Comment:
Note: I would need to get the app builder in the service and this is the only way I am able to.